src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package org.folio.innreach.domain.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.folio.innreach.domain.dto.InnReachTransactionItemHoldDTO;
+import org.folio.innreach.domain.entity.InnReachTransaction;
+import org.folio.innreach.domain.entity.InnReachTransaction.TransactionType;
+import org.folio.innreach.domain.service.InnReachTransactionService;
+import org.folio.innreach.dto.TransactionItemHoldDTO;
+import org.folio.innreach.mapper.InnReachTransactionMapper;
+import org.folio.innreach.repository.InnReachTransactionRepository;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class InnReachTransactionServiceImpl implements InnReachTransactionService {
+
+  private final InnReachTransactionRepository repository;
+  private final InnReachTransactionMapper mapper;
+
+  @Override
+  public InnReachTransactionItemHoldDTO createInnReachTransactionItemHold(String trackingId, String centralCode, TransactionItemHoldDTO dto) {
+    InnReachTransaction transaction = new InnReachTransaction();
+    transaction.setTrackingId(trackingId);
+    transaction.setCentralServerCode(centralCode);
```

Comment:
shouldn't we validate if `centralCode` matches any saved code in `central_server.local_server_code` ?

Reply:
UPD it should be validated against `central_server.central_server_code`